autogpt_platform/backend/backend/blocks/google/_auth.py

```diff
@@ -0,0 +1,53 @@
+from typing import Literal
+
+from autogpt_libs.supabase_integration_credentials_store.types import OAuth2Credentials
+from pydantic import SecretStr
+
+from backend.data.model import CredentialsField, CredentialsMetaInput
+from backend.util.settings import Secrets
+
+secrets = Secrets()
+GOOGLE_OAUTH_IS_CONFIGURED = bool(
+    secrets.google_client_id and secrets.google_client_secret
+)
+
+GoogleCredentials = OAuth2Credentials
+GoogleCredentialsInput = CredentialsMetaInput[Literal["google"], Literal["oauth2"]]
+
+
+def GoogleCredentialsField(scopes: list[str]) -> GoogleCredentialsInput:
+    """
+    Creates a Google credentials input on a block.
+
+    Params:
+        scopes: The authorization scopes needed for the block to work.
+    """
+    return CredentialsField(
+        provider="google",
+        supported_credential_types={"oauth2"} if GOOGLE_OAUTH_IS_CONFIGURED else set(),
```

Comment:
If `not GOOGLE_OAUTH_IS_CONFIGURED` all Google blocks should probably be disabled. I didn't consider having empty `supported_credential_types` as a valid case to implement for, so in that sense it's undefined behavior.

Reply:
is there a place to say hey if the credentials are not set, don't load?

Reply:
oh brilliant